opts/opts_test.go

```diff
@@ -308,32 +308,27 @@ func TestNamedMapOpts(t *testing.T) {
 
 func TestParseLink(t *testing.T) {
 	name, alias, err := ParseLink("name:alias")
-	if err != nil {
-		t.Fatalf("Expected not to error out on a valid name:alias format but got: %v", err)
-	}
-	if name != "name" {
-		t.Fatalf("Link name should have been name, got %s instead", name)
-	}
-	if alias != "alias" {
-		t.Fatalf("Link alias should have been alias, got %s instead", alias)
-	}
+	assert.Check(t, is.Nil(err))
```

Comment:
Ah, yes, I've been a bit on the fence on those. Perhaps it's because we don't use it like that too often (yet?) so I'm not used to it, but a casual read of this could lead to "are we asserting that an error _occurred_?" 

```go
assert.Check(t, err)
```

I know I had to check the actual implementation the first time I saw a construct like that (to check _what_ it was asserting), which is why I picked the more verbose `is.Nil()`.

Maybe it's just me though, and I just need to get used to it; also happy to hear what others think, so that we can consistently pick an approach.

/cc @tianon @vvoland @cpuguy83

Reply:
Looks like we're already using this in various spots, so I updated

Reply:
I agree that the `assert.Check(t, err)` is a bit confusing. I'm used to it now and it's just another quirk when writing Go code, so I'm fine with it 😄.
However, personally I would choose `assert.NilError` for such assertion.

Reply:
> personally I would choose assert.NilError for such assertion.

The downside of `assert.NilError()` is that it triggers a `t.Fatal()` if it fails; we try to continue the test to capture any other errors, hence using `assert.Check()` for these. (I definitely agree `assert.NilError()` is more readable though!)